### PR TITLE
feat(docker-push-to-registries): Added

### PR DESCRIPTION
**Pull request by ACoolmanTelicent**
Related: https://github.com/Telicent-io/telicent-ontology-app/pull/249

Best (naive) guess at removing AWS

# Testing

Will need to happen "live" once merged to main

### CODE REVIEW

File: .github/workflows/docker-push-to-registries.yml [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+          builder: ${{ steps.buildx.outputs.name }}
+          platforms: linux/amd64,linux/arm64
+          push: true
+          context: ${{ inputs.PATH }}
+          file: ${{ inputs.PATH }}/${{ inputs.DOCKERFILE }}
+          tags: ${{ steps.meta.outputs.tags }}
+          target: ${{ inputs.TARGET }}
+          build-args: |
+            NPM_TOKEN=${{ secrets.NPM_REGISTRY_TOKEN }}
+            ${{ inputs.BUILD_ARGS }}
```
Comment (davec504) [inline .github/workflows/docker-push-to-registries.yml:143]: Is this necessary when the libraries it will be using will be public?

**Review by davec504 — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-push-to-registries.yml [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+
+      - name: Prepare Docker Image Names and Tags
+        id: meta
+        uses: docker/metadata-action@v4
+        with:
+          # Defines the base image names in repository/name format
+          images: |
+            ${{ env.DOCKER_USERNAME }}/${{ inputs.APP_NAME_PREFIX }}${{ inputs.APP_NAME }}${{ inputs.IMAGE_SUFFIX }}
```
Comment (rob-telicent) [inline .github/workflows/docker-push-to-registries.yml:111]:
```suggestion
            telicent/${{ inputs.APP_NAME_PREFIX }}${{ inputs.APP_NAME }}${{ inputs.IMAGE_SUFFIX }}
```
The docker username doesn't match our org name

**Review by rob-telicent — CHANGES_REQUESTED** [review]
Just needs a lick of sanity on the docker repo name

**Comment by ACoolmanTelicent** [comment]
Whoops. NOthing has changed yet. Ignore this review for a few minutes

File: .github/workflows/docker-push-to-registries.yml [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+        type: string
+        default: main
+        description: |
+          Specifies the main branch for the repository, defaults to main.  If this build is on the configured main
+          branch then the latest tag will be added to the image, if the build is on another branch then the development
+          tag will be added to the image.
+    outputs:
+      aws_image: 
+        description: |
+          The versioned AWS ECR image tag that was created and published.
```
Comment (davec504) [inline .github/workflows/docker-push-to-registries.yml:64]:
```suggestion
          The versioned docker image tag that was created and published.
```

File: .github/workflows/docker-push-to-registries.yml [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+        with:
+          builder: buildx_buildkit_${{ steps.buildx.outputs.name }}0
+          cache-path: /tmp/buildkit-cache
+          cache-max-size: 3g
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
```
Comment (tatelicent) [inline .github/workflows/docker-push-to-registries.yml:98]: Note this should only be used for images which will be public, not for private ones

File: .github/workflows/docker-push-to-registries.yml [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+          # A latest tag, if and only if, this commit is on the main branch
+          # A development tag, if this commit is not on the main branch
+          tags: |
+            type=ref,event=tag
+            type=semver,pattern={{version}}
+            type=sha,format=long,prefix=
+            type=raw,value=${{ inputs.VERSION }}${{ inputs.VERSION_SUFFIX }}
+            type=raw,value=latest,enable=${{ github.ref_name == inputs.MAIN_BRANCH }}
+            type=raw,value=development,enable=${{ github.ref_name != inputs.MAIN_BRANCH}}
+
```
[the remark this answers is not in the document]
Comment (ACoolmanTelicent) [reply .github/workflows/docker-push-to-registries.yml:124]: Note: I don't know how labels/actions do/can fit into our deployment pipeline, so I'm happy to do whatever y'all think is best

**Comment by ACoolmanTelicent** [comment]
Hm.

So it seems like I accidentally pressed merge?

Surprising as  
1. I don't feel like I pressed anything (I literally say the PR turn purple as I was typing - maybe I hit a kb shortcut?)
2. This does not have any approvals.

Its not terrible, as the diff moves things forward.

But I'm investigating..

**Comment by ACoolmanTelicent** [comment]
Oh.... ok.

So it may be because I pushed commits to this branch... from my local, directly to `telicent-oss`, and github resolved this by closing this PR.

I guess on the correct assumption I'll just open the PR in `telicent-oss`.

**Pull request merged**